### ISSUE TITLE
Problem with redefining a language and using babel

### ISSUE THREAD

**Issue by oalbiez** [ElegantLaTeX/ElegantBook#122]
I am working on a french book. I redefined the translation for french using :

    \RequirePackage[french]{babel}
    \setlength\parindent{2em}
    \newcommand\figref[1]{\textbf{Figure}~\ref{#1}}
    \newcommand\tabref[1]{\textbf{Tableau}~\ref{#1}}
    \renewcommand{\chaptername}{Chapitre \thechapter}
    \newcommand{\authorname}{\textbf{auteur: }}
    \newcommand{\institutename}{\textbf{Institut: }}
    ...

When I use 

    \documentclass[12pt,mode=fancy,cite=authoryear,titlestyle=display,lang=fr]{elegantbook}

The generated document does not have the chapter number in french, it is only present in english when lang=en.

The only difference I see is in french I use babel, in english not. In spite of this, I am unable to correct the problem. Do you have any ideas?

**Comment by sikouhjw** [comment]
Please provide [MWE](https://tex.meta.stackexchange.com/questions/228/ive-just-been-asked-to-write-a-minimal-working-example-mwe-what-is-that).

**Comment by oalbiez** [comment]
\documentclass[12pt,mode=fancy,cite=authoryear,titlestyle=display]{elegantbook}
    \usepackage[utf8]{inputenc}
    \usepackage[french]{babel}

    % == Translation in french
    \renewcommand{\chaptername}{Chapitre \thechapter}

    \begin{document}
    \maketitle
    \chapter{Test}
    \end{document}

If you comment the line `\usepackage[french]{babel}` the translation is correct with the chapter number. With babel, the chapter number is missing.

**Comment by sikouhjw** [comment]
I can't solve this problem in my ability, sorry.

**Comment by EthanDeng** [comment]
The missing number is known problem in version 3, I have fix it in version 4, later in this year, version 4 will be released.

This problem is caused by different logic when dealing with the chapter counter and chapter name.

**Comment by gcorrad** [comment]
Yes I deal with the same problem in spanish, it's because babel use \{lang}chaptername instead of \chaptername. 
In spanish i use \spanishcheptername... why you dont try with \frenchchaptername... anyway I tried it with the Italian in the same way without success.

\renewcommand{\spanishchaptername}{Capítulo \thechapter}

**Comment by jbwoods** [comment]
Seeing the update from 17 september 2020, I presume this has been fixed in 4.0?
Recently I used in Overleaf the Elegant book template and ran into this problem. 
Can this template be updated?
Thanks.

**Comment by EthanDeng** [comment]
> Seeing the update from 17 september 2020, I presume this has been fixed in 4.0? Recently I used in Overleaf the Elegant book template and ran into this problem. Can this template be updated? Thanks.

The overleaf version is older version (3.x), I didn't upload the latest version to overleaft, you can check the lastest version on your local machine, I will upload 4.x to overleaft later.
